samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -107,13 +127,24 @@ def __init__(
         self._aws_region = aws_region
         self._aws_profile = aws_profile
 
+        # set the default container mode
+        if not container_mode:
+            container_mode = ContainerMode.COLD.value
+
+        self._warm_containers = container_mode != ContainerMode.COLD.value
```

Comment:
Will this behave as intended in case of **lazy** mode?

Reply:
lazy && warm modes are Warm modes :) .. the difference is the in warm mode will initialize all containers upfront, but in lazy mode the containers will be initialized when invoked